src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -2327,7 +2327,7 @@ void parametersBefore(MethodInformation methodInfo) {
                     String s = methodInfo.parameterTypes[j].getComponentType().getName();
                     if (methodInfo.criticalRegion || methodInfo.valueGetter || methodInfo.valueSetter ||
                             methodInfo.memberGetter || methodInfo.memberSetter) {
-                        out.println("(j" + s + "*)env->GetPrimitiveArrayCritical(arg" + j + ", NULL);");
+                        out.println("reinterpret_cast<" + typeName[0] + ">(env->GetPrimitiveArrayCritical(arg" + j + ", NULL));");
```

Comment:
Why do we need to change this here?

Reply:
Because the C++ compiler refuses to cast jchar * to char16_t * (or something along this line, I didn't save the exact error message at the moment). Anyway, its necessary for the unsigned short -> char16_t transision.

Reply:
Well, let's see what else we can do, because this doesn't look correct...

Reply:
In the lines below, `(" + typeName[0] + typeName[1] + ")` get used, so if that works let's do that

Reply:
You mean with or without reinterpret_cast?

Reply:
Without

Reply:
OK. It works (on my macos).